### PR TITLE
Fix #1857: Use js.Math.{abs,min,max} for Double and Float.

### LINKED ISSUE #1857: j.l.Math.{abs,min,max} do not handle correctly -0.0
This is true for the `Float` and `Double` variants. We should use the implementations in `js.Math` for those instead.

For `Int` and `Long`, the current implementation is correct and should be kept.

### PR DESCRIPTION
**Pull request by nicolasstucki**

### CODE REVIEW

**Comment by nicolasstucki** [comment]
Note: Will need rebase if #1856 is merged first, otherwise the other one will need to be rebased.

**Comment by sjrd** [comment]
LGTM

File: test-suite/src/test/scala/org/scalajs/testsuite/javalib/MathTest.scala [hunk 1]
```diff
@@ -15,6 +15,51 @@ object MathTest extends JasmineTest {
 
   describe("java.lang.Math") {
 
+    it("should respond to `abs`") {
+      expect(Math.abs(0)).toBe(0)
+      expect(Math.abs(42)).toBe(42)
+      expect(Math.abs(-42)).toBe(42)
+      expect(Math.abs(0.0)).toBe(0.0)
+      expect(Math.abs(-0.0)).toBe(0.0)
```
Comment (sjrd) [inline test-suite/src/test/scala/org/scalajs/testsuite/javalib/MathTest.scala]: You have to use `expect(Math.abs(-0.0).equals(0.0)).toBeTruthy` here. Same for the line about about `Math.abs(0.0)`

File: test-suite/src/test/scala/org/scalajs/testsuite/javalib/MathTest.scala [hunk 2]
```diff
@@ -15,6 +15,51 @@ object MathTest extends JasmineTest {
...
+    it("should respond to `max`") {
+      expect(Math.max(0, 0)).toBe(0)
+      expect(Math.max(0, 2)).toBe(2)
+      expect(Math.max(2, 0)).toBe(2)
+      expect(Math.max(0.0, 2.0)).toBe(2.0)
+      expect(Math.max(2.0, 0.0)).toBe(2.0)
+      expect(Math.max(0.0, 0.0)).toBe(0.0)
+      expect(Math.max(-0.0, 0.0)).toBe(0.0)
+      expect(Math.max(0.0, -0.0)).toBe(0.0)
+      expect(Math.max(-0.0, -0.0)).toBe(-0.0)
```
Comment (sjrd) [inline test-suite/src/test/scala/org/scalajs/testsuite/javalib/MathTest.scala]: Same for these four.

File: test-suite/src/test/scala/org/scalajs/testsuite/javalib/MathTest.scala [hunk 3]
```diff
@@ -15,6 +15,51 @@ object MathTest extends JasmineTest {
...
+    it("should respond to `min`") {
+      expect(Math.min(0, 0)).toBe(0)
+      expect(Math.min(0, 2)).toBe(0)
+      expect(Math.min(2, 0)).toBe(0)
+      expect(Math.min(0.0, 2.0)).toBe(0.0)
+      expect(Math.min(2.0, 0.0)).toBe(0.0)
+      expect(Math.min(0.0, 0.0)).toBe(0.0)
+      expect(Math.min(-0.0, 0.0)).toBe(-0.0)
+      expect(Math.min(0.0, -0.0)).toBe(-0.0)
+      expect(Math.min(-0.0, -0.0)).toBe(-0.0)
```
Comment (sjrd) [inline test-suite/src/test/scala/org/scalajs/testsuite/javalib/MathTest.scala]: And same again.

**Comment by sjrd** [comment]
That's all.

**Comment by sjrd** [comment]
LGTM

**Comment by sjrd** [comment]
Needs a rebase.

**Comment by nicolasstucki** [comment]
Jenkins, retest this please

**Comment by sjrd** [comment]
retest this please

**Comment by sjrd** [comment]
retest this please

**Comment by sjrd** [comment]
retest this please

**Comment by nicolasstucki** [comment]
retest this please

**Pull request merged**